src/compiler/transformers/es2015.ts

```diff
@@ -3338,7 +3338,7 @@ namespace ts {
         priority: 0,
         text: `
             var __extends = (this && this.__extends) || function (d, b) {
-                for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
+                Object.setPrototypeOf(d, b);

```

Comment:
We can use `d.__proto__ = b;`.
but it is not official spec in ES3 and ES5.
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/proto

I misunderstood, Object.setPrototypeOf is a specification from ES2015. not ES5...
We should not use `Object.setPrototypeOf`.

Is this code correct?
```
if(typeof Object.setPrototypeOf === "function") {
  Object.setPrototypeOf(d, b);
} else {
  d.__proto__ = b;
}
```